client/src/components/EmojiPicker.js

```diff
@@ -0,0 +1,169 @@
+'use strict';
+
+import _ from 'lodash';
+import React from 'react';
+import ReactEmoji from "react-emoji";
+import { emojiData } from "../utils/emojilist";
+import emojisAnnotations from 'emoji-annotation-to-unicode';
+import 'styles/EmojiPicker.scss';
+
+const modulo = (op1, op2) => (((op1 % op2) + op2) % op2);
+const supportedEmojis = Object.values(emojisAnnotations).filter((e) => e !== '1f642');
+const emojiList = _.pickBy(emojiData, (e) => supportedEmojis.indexOf(e.unicode) > -1);
+
+const filterEmojis = function(emojiList, word, amount) {
+  return Object.values(emojiList)
+    .filter((e) => e.shortname.indexOf(word) > -1 || e.aliases.indexOf(word) > -1 || word === '')
+    .slice(0, amount);
+};
+
+const emojify = (emojis, highlightedIndex, onMouseEnter, onClick) => {
+  const emojiOpts = { emojiType: 'emojione' };
+  return emojis
+    .map(e => ReactEmoji.emojify(e.shortname, emojiOpts))
+    .map((e, index) => <li key={index}
+                           onMouseEnter={ e => { e.preventDefault(); onMouseEnter(index); }}
+                           onClick={ e => { e.preventDefault(); onClick(index); }}
+                           className={index === highlightedIndex ? 'selected' : ''}>{e}</li>);
+};
```

Comment:
Add {} around the last .map and parenthesis around the onClick/onMouseEnter function arg. Makes it a little more readable imo.
```javascript
.map((e, index) => {
  return <li key={index}
             onMouseEnter={ (e) => { e.preventDefault(); onMouseEnter(index); }}
             onClick={ (e) => { e.preventDefault(); onClick(index); }}
             className={index === highlightedIndex ? 'selected' : ''}>{e}</li>
});

```